internal/provider/organization_data_source.go

```diff
@@ -90,12 +91,11 @@ func (d *OrganizationDataSource) Read(ctx context.Context, req datasource.ReadRe
 
 	// we fetch the organization id from the service account
 	orgMrn := ""
-	if data.OrgMrn.ValueString() != "" {
+	if data.OrgMrn.ValueString() != "" && data.OrgID.ValueString() == "" {
```

Comment:
Why is that extra check needed? When the mrn was provided, we should not be required to check for org id.